core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphqlActionBuilder.kt

```diff
@@ -0,0 +1,127 @@
+package org.evomaster.core.problem.graphql
+
+import graphql.language.*
+import graphql.schema.idl.TypeDefinitionRegistry
+import org.evomaster.core.problem.graphql.param.GraphqlParam
+import org.evomaster.core.problem.rest.param.Param
+import org.evomaster.core.search.Action
+import org.evomaster.core.search.gene.*
+import org.slf4j.LoggerFactory
+import org.slf4j.Logger
+import java.lang.IllegalStateException
+import java.util.concurrent.atomic.AtomicInteger
+
+
+class GraphqlActionBuilder {
+
+    companion object {
+        private val log: Logger = LoggerFactory.getLogger(GraphqlActionBuilder::class.java)
+        private val idGenerator = AtomicInteger()
+
+        fun addActionsFromSchema(schema: TypeDefinitionRegistry,
+                                 actionCluster: MutableMap<String, Action>) {
+
+            actionCluster.clear()
+
+            val rootOperations = mutableListOf<String>("Mutation", "Query")
+
+            rootOperations.forEach{
+                val obj = schema.getType(it).get() as ObjectTypeDefinition
+                val operationType = it
+                val operationList = obj.fieldDefinitions.forEach{
+
+                    val params = extractParams(it, schema)
+
+                    val action = GraphqlCallAction("${it.name}${idGenerator.incrementAndGet()}", it.name, params, OperationType.valueOf(operationType))
+
+                    actionCluster.put(action.getName(), action)
+                }
+            }
+        }
+
+
+        private fun extractParams(operation: FieldDefinition, schema: TypeDefinitionRegistry): MutableList<GraphqlParam> {
+                    val params: MutableList<GraphqlParam> = mutableListOf()
+                    val name = operation.name
+
+                    operation.inputValueDefinitions.forEach{
+                        val inputName = it.name
+
+                        val type: TypeName
+                        when {
+                            it.type is NonNullType ->
```

Comment:
if all the checks are on `it.type`, could simply have it directly in the `when`, eg `when(it.type){...`